conans/client/cmd/export.py

```diff
@@ -233,6 +233,10 @@ def _replace_scm_data_in_conanfile(conanfile_path, scm_data):
                         comments = [line.strip('\n') for line in replace
                                     if line.strip().startswith("#") or not line.strip()]
                         break
+
+    if len(to_replace) == 0:
+        # SCM exists, but not found in the conanfile, probably inherited from superclass
+        return
```

Comment:
Here we need to actually add the `scm_data` to this `conanfile.py` file. There is nothing to replace, but we need to add the dictionary to the class.

Reply:
you are right.